src/main/java/io/hhplus/tdd/point/service/PointService.java

```diff
@@ -0,0 +1,75 @@
+package io.hhplus.tdd.point.service;
+
+import io.hhplus.tdd.database.PointHistoryTable;
+import io.hhplus.tdd.database.UserPointTable;
+import io.hhplus.tdd.point.entity.PointHistory;
+import io.hhplus.tdd.point.entity.UserPoint;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.locks.Lock;
+
+import io.hhplus.tdd.lock.LockManager;
+
+import static io.hhplus.tdd.point.entity.TransactionType.*;
+
+@Service
+@RequiredArgsConstructor
+public class PointService {
```

Comment:
```suggestion
public class PointService implements PointUseCase {
```

상위에 `PointUseCase`  또는 `IPointService` 와 같은 인터페이스를 선언 후 

Service 레이어에서 `PointService` 를 구현한다면

역할 기반 설계, 테스트 용이성, 관심사 분리 등 이점이 더 많을것으로 보입니다 👍